### PR TITLE
Create Attendees item row for editing requests

### PR DESCRIPTION
**Pull request by SzymczakJ**
### Details

This Pr adds option to edit attendees n a money request. Keep in mind that request for editing and requesting money with attendees is not working yet, but with onyx optimistic data everything should be easy to test.

### Fixed Issues

$ https://github.com/Expensify/App/issues/47173#issuecomment-2343055095
PROPOSAL:

### Tests

1. Create workspace with control plan.
2. Submit expense to this workspace and try adding attendees.
3. On created request try modifying attendees, verify that proper report action message is added under the request.
4. Submit expense that exceeds amount limit by small factor(amount limit is typically $2000) and keep only one attendee.
5. Add one attendee to this request and check, if violation "over limit" disappears.

### QA Steps

Same as tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/a32f929e-7739-47fa-91b3-21a4b6b4fc9c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/122e9218-f2df-4785-8684-df26845b900f

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/561121cd-fec4-4016-840c-85f0a1b7623c

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/595b837b-1c4e-4eda-8575-a72c148c981f

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/0d4cad47-e07a-4888-8b48-763772d57864

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/79c9071d-a377-4d48-a635-d6ed275c1fd7

</details>

### CODE REVIEW

File: src/pages/iou/request/step/IOURequestStepAttendees.tsx [hunk 1]
```diff
@@ -27,22 +28,24 @@ type IOURequestStepAttendeesOnyxProps = {
     policyTags: OnyxEntry<OnyxTypes.PolicyTagLists>;
 };
 
-type IOURequestStepAttendeesProps = IOURequestStepAttendeesOnyxProps & WithWritableReportOrNotFoundProps<typeof SCREENS.MONEY_REQUEST.STEP_ATTENDEES>;
+type IOURequestStepAttendeesProps = IOURequestStepAttendeesOnyxProps &
+    WithWritableReportOrNotFoundProps<typeof SCREENS.MONEY_REQUEST.STEP_ATTENDEES> &
+    WithFullTransactionOrNotFoundProps<typeof SCREENS.MONEY_REQUEST.STEP_ATTENDEES>;
```
Comment (zfurtak) [inline src/pages/iou/request/step/IOURequestStepAttendees.tsx:33]: I think in the end this is not needed 😅

File: src/pages/iou/request/step/withFullTransactionOrNotFound.tsx [hunk 2]
```diff
@@ -22,6 +22,7 @@ type MoneyRequestRouteName =
     | typeof SCREENS.MONEY_REQUEST.CREATE
     | typeof SCREENS.MONEY_REQUEST.STEP_DISTANCE
     | typeof SCREENS.MONEY_REQUEST.STEP_AMOUNT
+    | typeof SCREENS.MONEY_REQUEST.STEP_ATTENDEES
```
Comment (zfurtak) [inline src/pages/iou/request/step/withFullTransactionOrNotFound.tsx:25]: So here also, it's not needed

**Review by zfurtak — COMMENTED** [review]
Great! Just fix this one typescript error and we're good to go with this PR 😃

**Review by zfurtak — APPROVED** [review]
LGTM 🚀

**Comment by SzymczakJ** [comment]
@ZhenjaHorbach since you have full context in attendees feature and @allgandalf is probably not around, would you like to be C+ for this issue? 🙏

**Comment by Julesssss** [comment]
@allgandalf ah yes, sorry I didn't notice the auto-assignment here.

**Comment by Julesssss** [comment]
@ZhenjaHorbach please continue and let me know when the checklist is looking good 👍

**Comment by zfurtak** [comment]
@ZhenjaHorbach Kuba is a bit busy with other task so I updated the branch and adjusted to your comments 😉

File: src/libs/Violations/ViolationsUtils.ts [hunk 3]
```diff
@@ -345,6 +345,11 @@ const ViolationsUtils = {
                 return violation.name as never;
         }
     },
+
+    // We have to use regex, because Violation limit is given in a inconvenient form: "$2,000.00"
+    getViolationAmountLimit(violation: TransactionViolation): number {
```
[the remark this answers is not in the document]
Comment (zfurtak) [reply src/libs/Violations/ViolationsUtils.ts:350]: Okay, I will use it but maybe I'll leave the const for the regex, what do you think?

[the remark this answers is not in the document]
Comment (zfurtak) [reply src/libs/Violations/ViolationsUtils.ts:350]: Then it's done

**Review by ZhenjaHorbach — APPROVED** [review]

**Comment by zfurtak** [comment]
@Julesssss could you take a look at this one?

File: src/libs/Violations/ViolationsUtils.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Julesssss) [reply src/libs/Violations/ViolationsUtils.ts:350]: Agree with leaving the regex 👍

[the remark this answers is not in the document]
Comment (zfurtak) [reply src/libs/Violations/ViolationsUtils.ts:350]: @Julesssss then, could you review the PR and merge it of everything is right? 😃

**Review by Julesssss — APPROVED** [review]

File: src/libs/ReportUtils.ts [hunk 4]
```diff
@@ -3469,7 +3471,7 @@ function getModifiedExpenseOriginalMessage(
         originalMessage.merchant = transactionChanges?.merchant;
     }
     if ('attendees' in transactionChanges) {
-        [originalMessage.oldAttendees, originalMessage.attendees] = TransactionUtils.getFormattedAttendees(oldTransaction?.modifiedAttendees, oldTransaction?.attendees);
+        [originalMessage.oldAttendees, originalMessage.attendees] = TransactionUtils.getFormattedAttendees(transactionChanges?.attendees, TransactionUtils.getAttendees(oldTransaction));
```
Comment (Krishna2323) [inline src/libs/ReportUtils.ts:3474]: This line caused #75653, `oldAttendees` used `getAttendees(oldTransaction)` (includes default self-attendee), so first edit wrongly showed "changed" not "set". 

Fixed in #86259.

**Pull request merged**